app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,213 @@
+        alert.show()
+        val yes = alert.getButton(DialogInterface.BUTTON_POSITIVE)
+        val no = alert.getButton(DialogInterface.BUTTON_NEGATIVE)
+        yes.setTextColor(resources.getColor(R.color.md_blue_500))
+        no.setTextColor(resources.getColor(R.color.md_red_500))
+    }
+
+    override fun alertSuccess() {
+        val dialogClickListener = DialogInterface.OnClickListener { _, _ -> finish() }
+        val alertTitle = getString(R.string.signup)
+        val alertMessage = getString(R.string.signup_msg)
+        val successAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, null, resources.getString(R.string.ok), null, resources.getColor(R.color.md_blue_500))
+        successAlertboxHelper.showAlertBox()
+    }
+
+    override fun alertFailure() {
+        val dialogClickListener = DialogInterface.OnClickListener { _, _ ->
+            val intent = Intent(this@SignUpActivity, LoginActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+            startActivity(intent)
+            finish()
+        }
+        val dialogClickListenern = DialogInterface.OnClickListener { _, _ ->
+            val intent = Intent(this@SignUpActivity, ForgotPasswordActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+        val alertTitle = getString(R.string.error_email)
+        val alertMessage = getString(R.string.error_msg)
+        val failureAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, dialogClickListenern, resources.getString(R.string.ok), resources.getString(R.string.forgot_pass_activity), resources.getColor(R.color.md_blue_500))
+        failureAlertboxHelper.showAlertBox()
+    }
+
+    override fun alertError(message: String) {
+        val errorAlertboxHelper = AlertboxHelper(this@SignUpActivity, getString(R.string.unknown_host_exception), message, null, null, resources.getString(R.string.ok), null, Color.BLUE)
+        errorAlertboxHelper.showAlertBox()
+    }
+
+    override fun setErrorEmail() {
+        email?.error = getString(R.string.invalid_email)
+    }
+
+    override fun setErrorPass() {
+        password?.error = getString(R.string.error_password_matching)
+    }
+
+    override fun setErrorConpass(msg: String) {
+        confirm_password?.error = msg
+    }
+
+    override fun setErrorUrl() {
+        input_url?.error = getString(R.string.invalid_url)
+    }
+
+    override fun enableSignUp(bool: Boolean) {
+        sign_up?.isEnabled = bool
+    }
+
+    override fun clearField() {
+        CredentialHelper.clearFields(email, password, confirm_password)
+    }
+
+    override fun setupPasswordWatcher() {
+        password?.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                CredentialHelper.isPasswordValid(password.editText?.text.toString())
```

Comment:
@mayank408 Just one small last thing. I did not find usage of this method. This method is never called.  Also
`CredentialHelper.isPasswordValid(password.editText?.text.toString())` this line just returns boolean which is not used. Also onFocusChangeListener is applied on edit text. So according me it should be changed to 
```
//remove override and call from oncreate
fun setupPasswordWatcher() {
    // change password to password.editText
    password.editText?.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
    password.error = null
    if (!hasFocus) 
        presenter.checkForPassword()
    }
}
```
in presenter
```
override fun checkForPassword() {
if(!interactor.isvalidPassword())
   signupView.passwordInvalid()
}
```
and in interactor write 
```
override fun isvalidPassword() {
return CredentialHelper.isPasswordValid(password.editText?.text.toString())
}
```